safe_grid_agents/spiky/agents.py

```diff
@@ -83,6 +86,22 @@ def _get_TLV(self, boardX, rewardX, state_iterator) -> float:
                 unique_states.add(boardY.tostring())
         return TLV
 
+    def _purge_memory(self) -> None:
```

Comment:
This looks good to me.  One observation is that this no longer conforms to the algorithm in our paper -- I'm comfortable including it as an approximate solution if we can back it up with our existing experiments, but have you put any additional thought into how this fits into our theory?

Reply:
Since we only use states encountered in a single trajectory for identifying corrupt states, this change only affects the lower bound computation. We might get a slightly worse lower bound only if we encountered a state that would provide a better one before we identified a corrupt state, but dropped it and never encountered it again. That seems very unlikely. I think we can even expect most states that improve the lower bound to be encountered in the same trajectory as the corrupt state.